File: training-data/rust/coin_change.rs
fn coin_change(coins: &[i32], amount: usize) -> i32 {
    let mut dp = vec![i32::MAX; amount + 1];
    dp[0] = 0;
    for a in 1..=amount {
        for &coin in coins {
            if coin as usize <= a && dp[a - coin as usize] != i32::MAX {
                dp[a] = dp[a].min(dp[a - coin as usize] + 1);
            }
        }
    }
    if dp[amount] == i32::MAX {
        -1
    } else {
        dp[amount]
    }
}

fn main() {
    println!("{}", coin_change(&[1, 2, 5], 11));
}

File: training-data/c/coin_change.c
#include <stdio.h>

int coin_change(int coins[], int n, int amount) {
    int dp[12];
    for (int i = 0; i <= amount; i++) dp[i] = amount + 1;
    dp[0] = 0;
    for (int a = 1; a <= amount; a++) {
        for (int c = 0; c < n; c++) {
            if (coins[c] <= a && dp[a - coins[c]] + 1 < dp[a]) {
                dp[a] = dp[a - coins[c]] + 1;
            }
        }
    }
    return dp[amount] > amount ? -1 : dp[amount];
}

int main(void) {
    int coins[] = {1, 2, 5};
    printf("%d\n", coin_change(coins, 3, 11));
    return 0;
}
